libs/extensions/src/settings/settings.service.ts

```diff
@@ -81,16 +82,48 @@ export class SettingsService {
     return publicSetting;
   }
 
-  async listPublic() {
-    const publicSettings = await this.prisma.setting.findMany({
-      where: { isPrivate: false },
-    });
-    const result: any = {};
-    for (const setting of publicSettings) {
-      result[setting.name] = setting.value;
+  async listPublic(query: ListSettingDto) {
+    // const publicSettings = await this.prisma.setting.findMany({
+    //   where: { isPrivate: false },
+    // });
+    // const result: any = {};
+    // for (const setting of publicSettings) {
+    //   result[setting.name] = setting.value;
+    // }
+    // console.log({ result });
+    // return result;
+
+    const baseCondition = { isPrivate: false };
+    const AND_CONDITIONS = [];
+    let conditions = {};
+
+    if (query.name) {
+      AND_CONDITIONS.push({
+        name: { contains: query.name, mode: 'insensitive' },
+      });
+      conditions = { AND: AND_CONDITIONS };
     }
-    console.log({ result });
-    return result;
+
+    const select: Prisma.SettingSelect = {
+      name: true,
+      dataType: true,
+      isPrivate: true,
```

Comment:
I don't think we need to show this field since we are only listing public settings.

I think we can have the API to list all the settings but we should only return the private settings Name but not value.
cc: @santosh-rumsan @binod-rumsan

Reply:
@manjik-rumsan ,  private seetings value are protected and public value are shown